examples/zkquiz/contracts/src/VerifierContract.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.12;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {ERC721URIStorage} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+
+contract VerifierContract is ERC721URIStorage {
+    uint256 private _nextTokenId;
+
+    address public alignedServiceManager;
+
+    bytes32 public elfCommitment = 0x35dd40ab04e180712996495caec915b8a7c488433acbb50c4d8d912cb55bf1f1;
+
+    // map to check if proof has already been submitted
+    mapping(bytes32 => bool) public mintedProofs;
+
+    constructor(address _alignedServiceManager) ERC721("Aligned Layer ZK Quiz", "AZKQ") {
+        alignedServiceManager = _alignedServiceManager;
+    }
+
+    function verifyBatchInclusion(
+        bytes32 proofCommitment,
+        bytes32 pubInputCommitment,
+        bytes32 provingSystemAuxDataCommitment,
+        bytes20 proofGeneratorAddr,
+        bytes32 batchMerkleRoot,
+        bytes memory merkleProof,
+        uint256 verificationDataBatchIndex
+    ) external returns (uint256) {
```

Comment:
Just curious, did you try inheriting our VerifyBatchInclusionCaller contract? Might be simpler, instead of copypasting its code